### DISCUSSION TITLE
Override app environment variables per server

### DISCUSSION THREAD

**Discussion by jabczyk** [coollabsio/coolify#3065] [Feature Requests]
I would like to be able to override certain environment variables of my app depending on which server it's running on.

## Possible use cases
a) When hosting both front-end and back-end on multiple servers, it may be beneficial to let front-end talk to its own back-end directly, instead of going through a load balancer
b) Passing server name as environment variable to simplify monitoring and debugging, eg. in logs

## Possible solution
I suggest all servers share the same environment variables, just like now. On top of this, there's a separate list of variables for each server that override the defaults.

**Comment by pkpio** [comment]
This is very useful for my use case as well. I have a worker app deployed on multiple servers. Each job keeps track of the machine it ran on for debugging and health stats. 

Without the ability to set a server specific variable, we see the same server for all jobs. Only workaround is to deploy worker application twice - once per server.
